src/engraving/dom/textbase.cpp

```diff
@@ -3370,11 +3392,36 @@ void TextBase::undoChangeProperty(Pid id, const PropertyValue& v, PropertyFlags
             undoChangeProperty(Pid::TEXT, stripText(false, false, true), propertyFlags(id));
         }
     }
-    if (id == Pid::FONT_STYLE || id == Pid::FONT_FACE || id == Pid::FONT_SIZE || id == Pid::TEXT_SCRIPT_ALIGN) {
-        // can't use standard change property as Undo might set to "undefined"
-        score()->undo(new ChangeTextProperties(m_cursor, id, v, ps));
-    } else {
+
+    bool isTextSpecificProperty = id == Pid::FONT_STYLE || id == Pid::FONT_FACE || id == Pid::FONT_SIZE || id == Pid::TEXT_SCRIPT_ALIGN;
+    if (!isTextSpecificProperty) {
         EngravingItem::undoChangeProperty(id, v, ps);
+        return;
+    }
+
+    const std::list<EngravingObject*> linkedObjects = linkListForPropertyPropagation();
+    for (EngravingObject* linkedObject : linkedObjects) {
+        TextBase* linkedText = toTextBase(linkedObject);
+        if (linkedText == this) {
+            // can't use standard change property as Undo might set to "undefined"
+            score()->undo(new ChangeTextProperties(m_cursor, id, v, ps));
+            continue;
+        }
+
+        Score* linkedScore = linkedText->score();
+        TextCursor* linkedCursor = linkedText->cursor();
```

Comment:
Am I understanding correctly that this is about setting properties to selected characters of the text (and that _that_ is the reason that TextCursor is being used here)? In that case, can we be certain that the `linkedCursor` is synchronised with the "original" one? If it's not, I'm afraid that could give quite strange results, couldn't it?

Reply:
No, this gets called also when changing the property of the "full" text item, without necessarily entering edit mode and therefore having an actual cursor. `linkedCursor` is a default-initialized one, it's not actually linked to the present cursor. When editing individual characters things get more complicated and indeed there are still imperfections, but that's a minor case that can (and will) be smoothed out later. The format of the full text works fine as expected